src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-views/use-psbt-decoded-request.tsx

```diff
@@ -0,0 +1,110 @@
+import { getAddressFromOutScript } from '@shared/crypto/bitcoin/bitcoin.utils';
+import { logger } from '@shared/logger';
+import { isUndefined } from '@shared/utils';
+
+import {
+  OrdApiInscriptionTxOutput,
+  useOrdinalsAwareUtxoQueries,
+} from '@app/query/bitcoin/ordinals/ordinals-aware-utxo.query';
+import { useCurrentBtcNativeSegwitAccountAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
+
+import { buildPsbtInputsForUi } from './build-psbt-input-for-ui';
+import { matchInputsOutputs } from './match-inputs-and-outputs';
+
+interface NonWitnessUtxo {
+  version: number;
+  segwitFlag: boolean;
+  inputs: btc.TransactionInputRequired[];
+  outputs: btc.TransactionOutputRequired[];
+  witnesses: string[][];
+  lockTime: number;
+}
+
+interface WitnessUtxo {
+  script: Uint8Array;
+  amount: bigint;
+}
+
+export interface PsbtInput {
+  nonWitnessUtxo?: NonWitnessUtxo;
+  witnessUtxo?: WitnessUtxo;
+}
+
+export interface PsbtInputForUi extends PsbtInput {
+  address: string;
+  unsignedUtxo?: OrdApiInscriptionTxOutput;
+  value: number;
+}
+
+function isPlaceholderTransaction(
+  address: string,
+  inputs: (OrdApiInscriptionTxOutput | undefined)[],
+  outputs: btc.TransactionOutputRequired[],
+  network: BitcoinNetworkModes
+) {
+  const inputsNotFromCurrentAddress = inputs.filter(input => {
+    return input?.address !== address;
+  });
+  const outputsNotToCurrentAddress = outputs.filter(output => {
+    const addressFromScript = getAddressFromOutScript(output.script, network);
+    return addressFromScript !== address;
+  });
+  return inputsNotFromCurrentAddress.length === 0 && outputsNotToCurrentAddress.length === 0;
+}
+
+interface UsePsbtDecodedRequestArgs {
+  psbtInputs: PsbtInput[];
+  unsignedInputs: btc.TransactionInputRequired[];
+  unsignedOutputs: btc.TransactionOutputRequired[];
+}
+export function usePsbtDecodedRequest({
+  psbtInputs,
+  unsignedInputs,
+  unsignedOutputs,
+}: UsePsbtDecodedRequestArgs) {
+  const [showPlaceholder, setShowPlaceholder] = useState(false);
+  const network = useCurrentNetwork();
+  const bitcoinAddressNativeSegwit = useCurrentBtcNativeSegwitAccountAddressIndexZero();
+  const unsignedUtxos = useOrdinalsAwareUtxoQueries(unsignedInputs).map(query => query.data);
+
+  const inputs = useMemo(() => {
+    if (isUndefined(unsignedUtxos)) {
+      logger.error('No UTXOs to sign');
+      return [];
+    }
+    return buildPsbtInputsForUi({
+      network: network.chain.bitcoin.network,
+      psbtInputs,
+      unsignedUtxos,
+    });
+  }, [network.chain.bitcoin.network, psbtInputs, unsignedUtxos]);
+
+  const { fee, inputOutputPairs } = useMemo(
+    () =>
+      matchInputsOutputs({
+        psbtInputs: inputs,
+        unsignedOutputs,
+      }),
+    [inputs, unsignedOutputs]
+  );
+
+  useEffect(() => {
+    if (
+      isPlaceholderTransaction(
```

Comment:
Seems odd to call this in an effect and set the state. Plus, if it were later false, would never change back?

Should we just return `isPlaceholderTransaction` in the object directly, as it's an expression?

Reply:
I thought I had to do this for the unsignedUtxos query to load, but let me check. Good point though, will change to toggle.